### PR TITLE
Netcoreapp3

### PR DESCRIPTION
**Pull request by KevinRansom**
Build and run tests with netcoreapp3.0.

1.   Update projects and build.ps1
2.   Execute CompilerAssert testcases inproc, using appdomain on desktop
      and AssemblyLoadComplex on netcore.
      This is useful, because we don't have to grovel around looking for dotnet.exe, generate the
      runtime.json and copy fsharp.core.
3.   Bizarrely netcoreapp3.0 has a different format for ArgumentExceptions
4.   Remove, exit n from test cases that run under FSharp.Compiler
5.   Use failwith rather than returning an exit code, because exiting takes 
      down the whole testing process.
6.   For literal tests print the value, otherwise there would be an error that main is empty.

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -22,6 +22,13 @@
     <add key="vs-impl" value="https://vside.myget.org/F/vs-impl/api/v3/index.json" />
     <add key="roslyn_concord" value="https://myget.org/F/roslyn_concord/api/v3/index.json" />
     <add key="devcore" value="https://vside.myget.org/F/devcore/api/v3/index.json" />
+
```
Comment (brettfo) [inline NuGet.config]: Can you add a comment that these can be removed once the 3.0 SDK ships?

File: eng/Build.ps1 [hunk 2]
```diff
@@ -309,7 +309,7 @@ try {
         Prepare-TempDir
         EnablePreviewSdks
 
-        # enable us to build netcoreapp2.1 binaries
+        # enable us to build netcoreapp3.0 binaries
```
Comment (brettfo) [inline eng/Build.ps1]: This comment shouldn't have been changed.

File: src/fsharp/FSharp.Compiler.LanguageServer/FSharp.Compiler.LanguageServer.fsproj [hunk 3]
```diff
@@ -5,7 +5,7 @@
     <OutputType>Exe</OutputType>
     <TargetExt>.exe</TargetExt>
     <TargetFrameworks>net472;netcoreapp2.1</TargetFrameworks>
```
Comment (brettfo) [inline src/fsharp/FSharp.Compiler.LanguageServer/FSharp.Compiler.LanguageServer.fsproj]: Change this line, too.

File: tests/fsharp/core/pinvoke/test.fsx [hunk 4]
```diff
@@ -16,6 +16,26 @@ let report_failure (s : string) =
     stderr.WriteLine s
     failures := !failures @ [s]
 
+// We currently build targeting netcoreapp2_1, and will continue to do so through this VS cycle
+// We will use this api to see if we are running on a netcore which supports pinvoke / refemit
```
Comment (brettfo) [inline tests/fsharp/core/pinvoke/test.fsx]: Nit: add link to API reference for this call.

File: tests/fsharp/Compiler/CompilerAssert.fs [hunk 5]
```diff
@@ -102,6 +106,37 @@ let main argv = 0"""
                 try Directory.Delete(projectDirectory) with | _ -> ()
 #endif
 
+#if FX_NO_APP_DOMAINS
```
Comment (TIHan) [inline tests/fsharp/Compiler/CompilerAssert.fs]: This is good. Means we can build this test project as any TFM without worrying about runtime configs and copying FSharp.Core/etc.

**Pull request merged**